lib/puppet/defaults.rb

```diff
@@ -985,31 +985,43 @@ def self.valid_file_checksum_types
       :owner => "service",
       :group => "service",
       :mode => "0755",
-      :desc => "The root directory for the certificate authority."
+      :desc => "The root directory for the certificate authority.",
+      :hook => proc { |value|
+        Puppet.deprecation_warning(_("The 'cadir' setting is deprecated. Starting in Puppet 6, this setting will be ignored in favor of the analagous setting in Puppet Server's config."))
+      }
     },
     :cacert => {
       :default => "$cadir/ca_crt.pem",
       :type => :file,
       :owner => "service",
       :group => "service",
       :mode => "0644",
-      :desc => "The CA certificate."
+      :desc => "The CA certificate.",
+      :hook => proc { |value|
+        Puppet.deprecation_warning(_("The 'cacert' setting is deprecated. Starting in Puppet 6, this setting will be ignored in favor of the analagous setting in Puppet Server's config."))
```

Comment:
Is this a thing where all of these become one translatable string if we pull out the setting name and interpolate it? (that's the case in clojure, not sure about ruby)

Reply:
Yeah I think so. And I think I just thought of a way to do it.